File: studio/src/main/java/br/com/api/enums/TypeOfSupplierEnum.java
package br.com.api.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum TypeOfSupplierEnum {

	MERCHANDISE("label.merchandise.supplier"), 
	ELETRICITY("label.eletricity.supplier"), 
	WATER("label.water.supplier"), 
	CARRYING("label.carrying.supplier"), 
	RENT("label.rent"), 
	OTHER("label.other");


	private final String i18n;
}

File: src/model/TypeOfSupplierEnum.ts
export enum TypeOfSupplierEnum {
  MERCHANDISE = 'Mercadoria',
  ELETRICITY = 'Eletricidade',
  WATER = 'Água',
  CARRYING = 'Transporte',
  RENT = 'Aluguel',
  OTHER = 'Outros'
}

export const TypeOfSupplierList: {
    key: string;
    value: string;
}[] = Object.entries(TypeOfSupplierEnum).map(([key, value]) => ({ key, value }))

export type TypeOfSupplierType = keyof typeof TypeOfSupplierEnum

export const TypeOfSupplierEntries = Object.entries(TypeOfSupplierEnum)
